CMakeLists.txt

```diff
@@ -1,283 +1,101 @@
-cmake_minimum_required(VERSION 3.10)
+cmake_minimum_required(VERSION 3.16...3.21)
```

Comment:
Why cap the CMake version at 3.21?

Reply:
It's the **minimum** required, not the maximum.

Reply:
I think it's in reference to the `...3.21` part, which according to the documentation is the "policy max", and not necessarily maximum version.

Reply:
Yeah the idea is to tell e.g. CMake 3.22 to use policies and features from version 3.16 _up to_ 3.21. This ensures that future CMake won't activate modern features that will break the scripts.